### PR TITLE
GitHub CI: Migrate CI to an Alpine container(GH removed Py27)

### PR DESCRIPTION
**Pull request by bernhardkaindl**
GitHub removed Python 2.7 from their Ubuntu 20.04 rootfs images and not all mitigations work for all cases (e.g. some don't work with the local GitHub CI test runner tool `act` to test actions):

Migrate tests/integration/sar-file-collection.test.sh to run in an alpine container with support for Python2.7. This also works for local CI using act and is *very* small *and* fast.

### CODE REVIEW

File: tests/integration/sar-file-collection.test.sh [hunk 1]
```diff
@@ -16,7 +16,7 @@ cp -a tests/integration/dom0-template/* /
 mkdir -p  /var/log/sa
 echo sa  >/var/log/sa/sa01
 echo sr  >/var/log/sa/sar31
-ln -sf    /bin/echo /bin/sar
+echo -e '#!/bin/sh\necho $*' >/bin/sar;chmod +x /bin/sar
```
Comment (lindig) [inline tests/integration/sar-file-collection.test.sh:19]: Why is this better than the sym link? Does `/bin/echo` not exist? I believe (but I'm not sure, `echo $@` is handling edge cases around quoting better.

**Review by lindig — APPROVED** [review]
This should be reviewed by recent committers.

File: tests/integration/sar-file-collection.test.sh (hunk 1 above)
Comment (lindig) [reply tests/integration/sar-file-collection.test.sh:19]:
```

Comment (bernhardkaindl) [reply tests/integration/sar-file-collection.test.sh:19]: @lindig:
> Why is this better than the sym link? Does /bin/echo not exist? 

In Alpine Linux, `echo` is a Busybox applet, and `/bin/echo` is a symlink to `busybox`:

* On execution of the Busybox applet, `busybox` reads its `"$0"` and based on that, `busybox` decides which code to run.

* Because with a symlink "$0" would be "/bin/sar", busybox would look for code to run for that, but that fails. Therefore, for `busybox echo` to be useable in this test, `echo` has to be invoked with `$0` set to `/bin/echo`.

> echo $@ is handling edge cases around quoting better.

Yes, but it is used a generated script to mock `/bin/sar`, and the sole purpose of that mock is to allow the test case to verify that the bugtool invoked `/bin/sar` with the expected arguments.

Because in this case, spaces and the grouping of arguments is not of significance, I intentionally omitted the use of "$@" to indicate for the reader that this mock script is just to echo back simple arguments and not an involved sequence of parameters contianing spaces, because for production code, not a mock, I would of course have used "$@".

So without spelling it out, I intented this to say (beween the lines): "This is a mock, not an actual user program".

Of course I can add a comment to explain all of this.

**Comment by lindig** [comment]
Can we merge this @bernhardkaindl?
